packages/api-providers/src/contracts/wrappers/webb-vanchor.ts

```diff
@@ -428,6 +432,75 @@ export class VAnchorContract {
     };
   }
 
+  // This function will query the chain for notes that are spendable by a keypair in a block range
+  async getSpendableUtxosFromChain(
+    owner: Keypair,
+    startingBlock: number,
+    finalBlock: number,
+    abortSignal: AbortSignal
+  ): Promise<Utxo[]> {
+    const filter = this._contract.filters.NewCommitment(null, null, null);
+    let logs: Array<Log> = []; // Read the stored logs into this variable
+
+    finalBlock = finalBlock || (await this.web3Provider.getBlockNumber());
+    console.log(`Getting notes from chain`);
+    // number of blocks to query at a time
+    const step = 1024;
+    console.log(`Fetching notes with steps of ${step} logs/request`);
+
+    try {
+      for (let i = startingBlock; i < finalBlock; i += step) {
+        const nextLogs = await retryPromise(
+          () => {
+            return this.web3Provider.getLogs({
+              fromBlock: i,
+              toBlock: finalBlock - i > step ? i + step : finalBlock,
+              ...filter,
+            });
+          },
+          20,
+          10,
+          abortSignal
+        );
+
+        logs = [...logs, ...nextLogs];
+
+        console.log(`Getting logs for block range: ${i} through ${i + step}`);
+      }
+    } catch (e) {
+      logger.error(e);
+      throw e;
+    }
+
+    const events = logs.map((log) => this._contract.interface.parseLog(log));
+    console.log(events[0]);
+
+    const encryptedCommitments: string[] = events.map((e) => e.args.encryptedOutput);
+
+    // Attempt to decrypt with the owner's keypair
+    const utxos = await Promise.all(
+      encryptedCommitments.map(async (enc) => {
+        try {
+          const utxo = await CircomUtxo.decrypt(owner, enc);
+          return utxo;
+        } catch (e) {
+          // eslint-disable-next-line no-empty
```

Comment:
Is this intentional?

Reply:
good point - updated to `return undefined`